src/NHibernate/Event/Default/DefaultLockEventListener.cs

```diff
@@ -42,7 +43,7 @@ public virtual void OnLock(LockEvent @event)
 			{
 				IEntityPersister persister = source.GetEntityPersister(@event.EntityName, entity);
 				object id = persister.GetIdentifier(entity);
-				if (!ForeignKeys.IsNotTransient(@event.EntityName, entity, false, source))
+				if (ForeignKeys.IsTransientFast(@event.EntityName, entity, source).GetValueOrDefault())
```

Comment:
`IsNotTransient` does additional check on `IsProxy` and `IsEntryFor`. Both of these checks are performed above at lines L31 and L41-L42.